test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1588 @@
+			framework.Logf("Pod is running on cgroupv2: " + stdout)
+			return true
+		}
+	}
+	framework.Logf("Assuming Pod is running on cgroupv2: " + stdout + "," + stderr)
+	return true
+}
+
+func verifyPodContainersCgroupValues(ctx context.Context, f *framework.Framework, pod *v1.Pod, tcInfo []TestContainerInfo, flagError bool) bool {
+	podOnCgroupv2Node := isPodOnCgroupv2Node(ctx, f, pod, true)
+	cgroupMemLimit := Cgroupv2MemLimit
+	cgroupCPULimit := Cgroupv2CPULimit
+	cgroupCPURequest := Cgroupv2CPURequest
+	if !podOnCgroupv2Node {
+		cgroupMemLimit = CgroupMemLimit
+		cgroupCPULimit = CgroupCPUQuota
+		cgroupCPURequest = CgroupCPUShares
+	}
+	verifyCgroupValue := func(cName, cgPath, expectedCgValue string) bool {
+		ginkgo.By("Sleep " + PollInterval.String() + " seconds")
+		time.Sleep(PollInterval)
+		mycmd := fmt.Sprintf("head -n 1 %s", cgPath)
+		cgValue, _, err := ExecCommandInContainerWithFullOutput(ctx, f, pod.Name, cName, "/bin/sh", "-c", mycmd)
+		framework.Logf("Namespace %s Pod %s Container %s - looking for cgroup value %s in path %s",
+			pod.Namespace, pod.Name, cName, expectedCgValue, cgPath)
+		if flagError {
+			framework.ExpectNoError(err, "failed to find expected value '%s' in container cgroup '%s'",
+				expectedCgValue, cgPath)
+		}
+		cgValue = strings.Trim(cgValue, "\n")
+		if flagError {
+			gomega.Expect(cgValue).Should(gomega.Equal(expectedCgValue), "cgroup value")
+		}
+		if cgValue != expectedCgValue {
+			return false
+		}
+		return true
+	}
+	for _, ci := range tcInfo {
+		if ci.Resources == nil {
+			continue
+		}
+		tc, _ := makeTestContainer(ci)
+		if tc.Resources.Limits != nil || tc.Resources.Requests != nil {
+			var cpuShares int64
+			var cpuLimitString, memLimitString string
+			memLimitInBytes := tc.Resources.Limits.Memory().Value()
+			cpuRequest := tc.Resources.Requests.Cpu()
+			cpuLimit := tc.Resources.Limits.Cpu()
+			if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+				cpuShares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
+			} else {
+				cpuShares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
+			}
+			cpuQuota := kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
+			if cpuLimit.IsZero() {
+				cpuQuota = -1
+			}
+			cpuLimitString = strconv.FormatInt(cpuQuota, 10)
+			if podOnCgroupv2Node {
+				if cpuLimitString == "-1" {
+					cpuLimitString = "max"
+				}
+				cpuLimitString = fmt.Sprintf("%s %s", cpuLimitString, CPUPeriod)
+			}
+			memLimitString = strconv.FormatInt(memLimitInBytes, 10)
+			if podOnCgroupv2Node && memLimitString == "0" {
+				memLimitString = "max"
+			}
+			if memLimitString != "0" {
+				if !verifyCgroupValue(ci.Name, cgroupMemLimit, memLimitString) {
+					return false
+				}
+			}
+			if !verifyCgroupValue(ci.Name, cgroupCPULimit, cpuLimitString) {
+				return false
+			}
+			if podOnCgroupv2Node {
+				// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
+				cpuShares = int64(1 + ((cpuShares-2)*9999)/262142)
```

Comment:
where'd you get this from? Can you link to a refence here?

Reply:
Done